skepticoin/datatypes.py

```diff
@@ -129,6 +129,7 @@ def __init__(self, inputs: List[Input], outputs: List[Output]):
         self.version = 0  # reserved for future use; the class does not take this as a param.
         self.inputs = inputs
         self.outputs = outputs
+        self.cached_hash = sha256d(self.serialize())
```

Comment:
This approach hinges on the immutability of transactions. In practice I think I've been smart enough to never treat transactions as mutable objects, but I know for a fact the immutability is never enforced. Since this particular line really creates a hard dependency on immutability, it's probably for the best to at least note that explicitly in a comment or even in the Transaction's docstring.

Reply:
Maybe mypy help with marking this variable as immutable?

Reply:
mypy helps ... 07b92324b80fe36effa4c69df1cfc571a82e5c12